reactor-core/src/test/java/reactor/core/publisher/FluxMetricsTest.java

```diff
@@ -456,4 +459,29 @@ public void flowDurationTagsConsistency() {
 
 		assertThat(uniqueTagKeySets).hasSize(1);
 	}
+
+	@Test
+	public void ensureFuseablePropagateOnComplete_inCaseOfAsyncFusion() {
+		Flux.fromIterable(Arrays.asList(1, 2, 3))
+		    .metrics()
+		    .flatMapIterable(Arrays::asList)
+		    .as(StepVerifier::create)
+		    .expectNext(1, 2, 3)
+		    .expectComplete()
+		    .verify(Duration.ofMillis(1));
```

Comment:
the `Duration` in the `verify` method is an upper bound after which the test times out. `1ms` is a bit short, as it could hiccup (eg. on CI machines). I suggest you raise it to `Duration.ofSeconds(1)` everywhere just to be on the safe side